Source/MLX/MLXArray+Indexing.swift

```diff
@@ -521,4 +353,1209 @@ extension MLXArray {
         }
     }
 
+    /// General array indexing with encoded operations.
+    ///
+    /// Typically used via the general array indexing subscript:
+    ///
+    /// ```swift
+    /// array[0, .ellipsis, 0 ..< 5]
+    /// ```
+    ///
+    /// ### See Also
+    /// - <doc:indexing>
+    subscript(
```

Comment:
This is the start of the new implementation.  This is an internal subscript operator that takes an array of `MLXArrayIndexOperation` to apply (the encoded operations).